src/request_handler.cc

```diff
@@ -1,111 +1,17 @@
-#include <string>
-#include <sstream>
-#include <boost/filesystem.hpp>
-#include <boost/filesystem/fstream.hpp>
 #include "request_handler.h"
 
-// todo: put each request handler into a separate file
+std::map<std::string, RequestHandler* (*)(void)>* request_handler_builders = nullptr;
 
-const std::string CONTENT_TYPE = "Content-Type";
-const std::string TEXT_PLAIN = "text/plain";
-const std::string STATIC_CONFIG_ROOT_KEY = "root";
+RequestHandler* RequestHandler::CreateByName(const char* type) {
+  const auto type_and_builder = request_handler_builders->find(type);
+  if (type_and_builder == request_handler_builders->end()) {
```

Comment:
This was copy/pasted from the profs. It doesn't matter if you use `count` or `end`, but I think that `count` is cleaner